supabase/migrations/20230315090547_remote_commit.sql

```diff
@@ -0,0 +1,1124 @@
+--
+
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp" WITH SCHEMA "extensions";
+
+
+--
+-- Name: update_buildings_latest_states(); Type: FUNCTION; Schema: public; Owner: postgres
+--
+
+CREATE FUNCTION "public"."update_buildings_latest_states"() RETURNS "trigger"
+    LANGUAGE "plpgsql" SECURITY DEFINER
+    AS $$BEGIN
+INSERT INTO buildings_latest_states (building_number, device_count, updated_at)
+SELECT COALESCE(new.building_number, old.building_number), SUM(device_count) as device_count_sum, MAX(updated_at) as updated_at_max
+FROM access_points_latest_states
+WHERE building_number = COALESCE(new.building_number, old.building_number) 
+--AND updated_at > (NOW() - '1 hour'::interval)
+ON CONFLICT (building_number)
+DO UPDATE SET device_count = EXCLUDED.device_count, updated_at = EXCLUDED.updated_at;
+RETURN NULL;
+END$$;
+
+
+ALTER FUNCTION "public"."update_buildings_latest_states"() OWNER TO "postgres";
+
+--
+-- Name: update_spaces_latest_states(); Type: FUNCTION; Schema: public; Owner: postgres
+--
+
+CREATE FUNCTION "public"."update_spaces_latest_states"() RETURNS "trigger"
+    LANGUAGE "plpgsql" SECURITY DEFINER
+    AS $$BEGIN
+IF COALESCE(new.room_id, old.room_id) IS NOT NULL THEN
+INSERT INTO spaces_latest_states (building_number, room_id, device_count, updated_at)
+SELECT COALESCE(new.building_number, old.building_number), COALESCE(new.room_id, old.room_id), SUM(device_count) as device_count_sum, MAX(updated_at) as updated_at_max
+FROM access_points_latest_states
+WHERE building_number = COALESCE(new.building_number, old.building_number) AND room_id = COALESCE(new.room_id, old.room_id)  --AND updated_at > (NOW() - '1 hour'::interval)
+ON CONFLICT (building_number, room_id)
+DO UPDATE SET device_count = EXCLUDED.device_count, updated_at = EXCLUDED.updated_at;
+END IF;
+RETURN NULL;
+END$$;
+
+
+ALTER FUNCTION "public"."update_spaces_latest_states"() OWNER TO "postgres";
+
+SET default_tablespace = '';
+
+SET default_table_access_method = "heap";
+
+--
+-- Name: access_points_latest_states; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE "public"."access_points_latest_states" (
+    "updated_at" timestamp with time zone NOT NULL,
+    "building_number" bigint,
+    "room_id" "text",
+    "device_count" bigint NOT NULL,
+    "floor" "text" NOT NULL,
+    "access_point_name" "text" NOT NULL,
+    "map_location" "text",
+    "location_hierarchy" "text"
+);
+
+
+ALTER TABLE "public"."access_points_latest_states" OWNER TO "postgres";
+
+--
+-- Name: buildings_latest_states; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE "public"."buildings_latest_states" (
+    "building_number" bigint NOT NULL,
+    "updated_at" timestamp with time zone NOT NULL,
+    "device_count" bigint NOT NULL
+);
+
+
+ALTER TABLE "public"."buildings_latest_states" OWNER TO "postgres";
+
+--
+-- Name: busy; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE "public"."busy" (
```

Comment:
I think we should delete this table and remove it from the migration script as it was just for a quick test.

Reply:
Talked with Misha about this we figured it could be a good test to delete it with a migration.